influxdb_client_3/__init__.py

```diff
@@ -194,10 +212,14 @@ def query(self, query, language="sql", mode="all", database=None,**kwargs ):
             database = self._database
         
         try:
-            headers = [(b"authorization", f"Bearer {self._token}".encode('utf-8'))]
-    
             # Create an authorization header
-            _options = FlightCallOptions(headers=headers, **kwargs)
+            args = {"headers": [(b"authorization", f"Bearer {self._token}".encode('utf-8'))]}
+            custom_args = {key: value for key, value in kwargs.items()}
```

Comment:
Could we make this conditional? Otherwise, we would need to run the header merge function on each query. I am wondering if we abstract this to the client class or not.  If not we could do an IF clause to check if the extra headers params are none and if so we just go with the default which I believe in most cases most people will.